### PR TITLE
Feature/basic web browsing

### PR DESCRIPTION
**Pull request by CDRussell**
Asana: https://app.asana.com/0/471012189846020/471193197990030

**Description**:
Basic implementation of a `WebView` with functioning `back/forward/refresh` buttons and the omnibar.

The UI is bare as designs are not finalised.

The omnibar entry differentiates between search queries and URLs. In both cases, it adds the required `t` and `tappv` query parameters.

**Steps to test this PR**:
1. Search for a simple query - "horse"
    - Ensure it performs a DDG search
2. Search for a more complex search - "4 + 5"
    - Ensure it performs the DDG search
3. Enter a URL - "https://example.com"
    - Ensure it loads the URL (and does not perform a search for it)
4. Repeat for more _exotic_ URL syntaxes

###### Author Checklist:
- [x] Get advice or leverage existing code
- [x] Agree on technical approach with reviewer (if the changes are nuanced)
- [x] Ensure that there is a testing strategy (and document any non-automated tests)
- [x] Consider implications (CPU, memory, database / preference migrations, statistics)

###### Reviewer Checklist:
- [x] **Ensure the PR solves the problem**
- [x] **Review every line of code**
- [x] **Ensure the PR does no harm by testing the changes thoroughly**
- [x] **Get help if you're uncomfortable with any of the above!**
- [x] Determine if there are any quick wins that improve the implementation

### CODE REVIEW

**Comment by subsymbolic** [comment]
One more obeservation, if I type "example.com" I'd like to be taken to http://example.com (https would be even better but https upgrades are complex so we'll cover that later)

File: app/src/main/java/com/duckduckgo/app/browser/DuckDuckGoRequestRewriter.kt [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+import android.webkit.WebResourceRequest
+import com.duckduckgo.app.browser.DuckDuckGoRequestRewriter.Constants.appVersionParam
+import com.duckduckgo.app.browser.DuckDuckGoRequestRewriter.Constants.sourceParam
+import com.duckduckgo.app.browser.omnibar.QueryUrlConverter
+import timber.log.Timber
+import javax.inject.Inject
+
+class DuckDuckGoRequestRewriter @Inject constructor() {
+
+    object Constants {
```
[the remark this answers is not in the document]
Comment (CDRussell) [reply app/src/main/java/com/duckduckgo/app/browser/DuckDuckGoRequestRewriter.kt]: I think ultimately the answer is _yes_.

Moreover, as well as marking them in a `companion` object, we should probably mark each with `const` or `private const` depending on if they're needed outside of this class or not, respectively.

[the remark this answers is not in the document]
Comment (CDRussell) [reply app/src/main/java/com/duckduckgo/app/browser/DuckDuckGoRequestRewriter.kt]: The alternative would be to forego the wrapper object altogether, and just declare them as `const` at a top level. ie, in the same file, but not inside the class per se.

From a brief search, the community seems divided on which approach is best. But i think the `companion` approach is cleaner. (I gather it _may_ generate an extra object to make this happen under the hood though)

**Comment by CDRussell** [comment]
@subsymbolic I've added the extra logic to load `http://example.com` from an input of `example.com`.

Also bumped the Kotlin version which just got released. 

Re-review plz

**Comment by subsymbolic** [comment]
All looks good 👍

**Pull request merged**